core/domain/rule_domain_test.py

```diff
@@ -98,10 +100,47 @@ def test_that_all_rules_have_object_editor_templates(self):
                     self.assertTrue(
                         param_obj_type.has_editor_js_template(),
                         msg='(%s)' % clazz.description)
-                    at_least_one_rule_found = True
-                clses.append(clazz)
 
-        self.assertTrue(at_least_one_rule_found)
+        self.assertEqual(EXPECTED_TOTAL_NUMBER_OF_RULES, num_rules)
```

Comment:
I think what the existence check mostly guards against is the following: suppose the rules system is refactored (as it is about to be). The hope is that this test will then error, since num_rules has gone from 47 to 0, and the developer will look at it, and think -- oh, I need to account for the types of the params and make sure that the spirit of this test gets replicated in whatever replaces the existing rules system.

I lean slightly towards keeping a check of some kind, since it's more likely that the rules system is going to change before a new rule is added. If that's an issue, though, we could fall back to the "> 0" version, since that would mean that creators of new rules won't have to change this test (and they really shouldn't have to).

I'm not too worried about finding the perfect solution here, though, since this whole file is likely to be short-lived (with reference to #1318).